src/app/address/AddressFormModal.tsx

```diff
@@ -0,0 +1,116 @@
+import { Country, FormField } from '@bigcommerce/checkout-sdk';
+import { withFormik, FormikProps } from 'formik';
+import React, { FunctionComponent } from 'react';
+import { lazy } from 'yup';
+
+import { preventDefault } from '../common/dom';
+import { withLanguage, TranslatedString, WithLanguageProps } from '../locale';
+import { Button, ButtonVariant } from '../ui/button';
+import { Form } from '../ui/form';
+import { LoadingOverlay } from '../ui/loading';
+import { Modal, ModalHeader } from '../ui/modal';
+
+import getAddressFormFieldsValidationSchema from './getAddressFormFieldsValidationSchema';
+import { AddressFormValues } from './mapAddressToFormValues';
+import AddressForm from './AddressForm';
+
+export interface AddressFormModalProps {
+    countries?: Country[];
+    countriesWithAutocomplete: string[];
+    googleMapsApiKey?: string;
+    isLoading: boolean;
+    shouldShowSaveAddress?: boolean;
+    isOpen: boolean;
+    defaultCountryCode?: string;
+    getFields(countryCode?: string): FormField[];
+    onSaveAddress(address: AddressFormValues): void;
+    onRequestClose?(): void;
+    onAfterOpen?(): void;
+}
+
+const AddressModal: FunctionComponent<AddressFormModalProps & WithLanguageProps & FormikProps<AddressFormValues>> = ({
+    googleMapsApiKey,
+    getFields,
+    isOpen,
+    onAfterOpen,
+    onRequestClose,
+    countriesWithAutocomplete,
+    countries,
+    values,
+    setFieldValue,
+    isLoading,
+}) => (
+    <Modal
+        additionalModalClassName="modal--medium"
+        header={
+            <ModalHeader>Add Address</ModalHeader>
+        }
+        isOpen={ isOpen }
+        onAfterOpen={ onAfterOpen }
+        onRequestClose={ onRequestClose }
+        shouldShowCloseButton={ true }
+    >
+        <Form autoComplete="on">
+            <LoadingOverlay isLoading={ isLoading }>
+                <AddressForm
+                    countries={ countries }
+                    countriesWithAutocomplete={ countriesWithAutocomplete }
+                    countryCode={ values.countryCode }
+                    formFields={ getFields(values.countryCode) }
+                    googleMapsApiKey={ googleMapsApiKey }
+                    setFieldValue={ setFieldValue }
+                    shouldShowSaveAddress={ false }
+                />
+                <div className="form-actions">
+                    <a
+                        className="button optimizedCheckout-buttonSecondary"
+                        href="#"
+                        onClick={ preventDefault(onRequestClose) }
+                    >
+                        <TranslatedString id="common.cancel_action" />
+                    </a>
+
+                    <Button
+                        disabled={ isLoading }
```

Comment:
Will `isLoading` be true when in the process of creating an address? I can't seem to find where the pending status is passed.

Reply:
yeah, it's currently being passed. It's the generic `isLoading` status from the SDK

Reply:
Oh okay. Are you sure though? 🤔  I tried to trace back the code and I found that `isLoading` is passed from `Shipping` component (https://github.com/bigcommerce/checkout-js/blob/master/src/app/shipping/Shipping.tsx#L300) and it only covers certain loading statuses, such as updating consignment etc... And creating an address is not in the list? But I might be reading the code wrong 😅

Reply:
I think I got confused with the loading spinner being showed up. I'm not actually checking for the isCreatingAddress flag as I'm currently doing that in the "background". I guess I could wait for both the consignment update and the address creation.